File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    let answer = new Array(id_list.length);
    answer.fill(0);
    let map = new Map();
    for(let i=0;i<id_list.length;i++){
        map.set(id_list[i],[]);
    }
    for(let i=0;i<report.length;i++){
        const re = report[i].split(' ');
        const tempData=map.get(re[1]);
        if(!tempData.find((e)=>e===re[0])){
            tempData.push(re[0]);
            map.set(re[1],tempData);
        }
    }
    map.forEach((data)=>{
        if(data.length>=k){
            for(let i=0;i<data.length;i++){
                const index = id_list.findIndex((e)=>e===data[i]);
                answer[index]++;
            }
        }
    })
    return answer;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <set>
#include <sstream>
#include <map>
#include <iostream>
using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    map<string,int> id_index;
    for(int i =0;i<id_list.size();i++){
        answer.push_back(0);
        id_index[id_list[i]]=i;
    }
    map<string,set<string>> report_list;
    for(int i = 0;i<report.size();i++){
        stringstream ss(report[i]);
        string first,second;
        ss>>first>>second;
        report_list[second].insert(first);
    }
    
    for(auto it : report_list ){
        if(it.second.size()>=k){
            for(auto name : it.second){
                answer[id_index[name]]++;
            }
        }
    }
    
    return answer;
}
/*
1. 신고를 당한 사람이 key 신고를 한사람이 value 
2. 신고를 한 사람이 k이상 이면 신고를 당한사람 목록에 있는 사람들의 인덱스 값에 1씩 추가
id_index{key : id , value : index}

*/